packages/core/src/StateNode.ts

```diff
@@ -358,89 +236,15 @@ class StateNode<
       this.type === 'final'
         ? (this.config as FinalStateNodeConfig<TContext, TEvent>).data
         : undefined;
-    this.invoke = toArray(this.config.invoke).map((invocable, i) => {
-      const id = `${this.id}:invocation[${i}]`;
-
-      const invokeConfig = toInvokeConfig(invocable, id);
-      const resolvedId = invokeConfig.id || id;
-
-      const resolvedSrc = isString(invokeConfig.src)
-        ? invokeConfig.src
-        : resolvedId;
-
-      if (
-        !this.machine.options.services[resolvedSrc] &&
-        !isString(invokeConfig.src)
-      ) {
-        this.machine.options.services = {
-          ...this.machine.options.services,
-          [resolvedSrc]: invokeConfig.src as any
-        };
-      }
-
-      return {
-        type: actionTypes.invoke,
-        ...invokeConfig,
-        src: resolvedSrc,
-        id: resolvedId
-      };
-    });
-
-    this.activities = toArray(this.invoke).map(toActivityDefinition);
-    this.transition = this.transition.bind(this);
-  }
-
-  private _init(): void {
-    if (this.__cache.transitions) {
-      return;
-    }
-    getAllStateNodes(this).forEach(stateNode => stateNode.on);
-  }
-
-  /**
-   * Clones this state machine with custom options and context.
-   *
-   * @param options Options (actions, guards, activities, services) to recursively merge with the existing options.
-   * @param context Custom context (will override predefined context)
-   */
-  public withConfig(
-    options: Partial<MachineOptions<TContext, TEvent>>,
-    context: TContext | undefined = this.context
-  ): StateNode<TContext, TStateSchema, TEvent> {
-    const { actions, activities, guards, services, delays } = this.options;
-
-    return new StateNode(
-      this.config,
-      {
-        actions: { ...actions, ...options.actions },
-        activities: { ...activities, ...options.activities },
-        guards: { ...guards, ...options.guards },
-        services: { ...services, ...options.services },
-        delays: { ...delays, ...options.delays }
-      },
-      context
-    );
-  }
-
-  /**
-   * Clones this state machine with custom context.
-   *
-   * @param context Custom context (will override predefined context, not recursive)
-   */
-  public withContext(
-    context: TContext
-  ): StateNode<TContext, TStateSchema, TEvent> {
-    return new StateNode(this.config, this.options, context);
   }
-
   /**
    * The well-structured state node definition.
    */
   public get definition(): StateNodeDefinition<TContext, TStateSchema, TEvent> {
     return {
       id: this.id,
       key: this.key,
-      version: this.version,
+      version: this.machine.version, // TODO: fix
```

Comment:
does this need any fixing? from what I see `this.machine` should always be defined and thus `this.machine.version` should just be available at all times